overlord/hookstate/hookmgr.go

```diff
@@ -307,10 +339,18 @@ func runHookAndWait(snapName string, revision snap.Revision, hookName, hookConte
 	// Hook completed; it may or may not have been successful.
 	case <-hookCompleted:
 		return buffer.Bytes(), hookError
-
-	// Hook was aborted.
+	// max timeout reached
+	case <-killTimerCh:
+		if err := killemAll(command); err != nil {
+			return nil, fmt.Errorf("cannot abort hook %q: %s", hookName, err)
+		}
+		return buffer.Bytes(), fmt.Errorf("exceeded maximum runtime of %s", maxRuntime)
```

Comment:
not 100% sure that that Bytes() is safe, you would need to loop around the select and wait for the hookCompleted or a different timer to check that we didn't spend too much time waiting for it to finish